cadl-extension/src/code-model-builder.ts

```diff
@@ -242,7 +243,7 @@ export class CodeModelBuilder {
       server.parameters.forEach((it) => {
         let parameter;
 
-        if (it.name === "ApiVersion") {
+        if (isApiVersion(this.program, it as any)) {
```

Comment:
This part I referred to [cadl-python](https://github.com/Azure/autorest.python/blob/main/packages/cadl-python/src/emitter.ts#L1143). I know that using `as any` is not a good practice, I saw Isabella created an issue in cadl-dpg, so I leave a question there https://github.com/Azure/cadl-azure/issues/2239#issuecomment-1398055201, I will keep our code updated with python-cadl.

Reply:
You belongs to the cadl-dpg

I would assume we modify it to take `HttpOperationParameter | ModelProperty`
https://github.com/Azure/cadl-azure/blob/fb8b612/packages/cadl-dpg/src/utils.ts#L38-L42

Reply:
I can take this issue as well. I will merge this pr first.